### PR TITLE
Blog permalink

### LINKED ISSUE #37: Add permalink to blog pages (only if on production)
If a page is a blog (if values.share=true or if parent.values.share=true and literalType=resource), add a link to the blog. If it's a blog parent (`values.share=true`), the link is https://qcs.shsbs.xyz/share/(page-hash). If the page is a blog child (`parent.values.share=true and literalType=resource`), the link is https://qcs.shsbs.xyz/share/(parent-hash)/(page-hash)

Uh this isn't really important, it'd just be nice to have and it's a super easy add if you have like some kind of infobox on the page with additional information.

### PR DESCRIPTION
**Pull request by darienreese**
resolves #37

image

### CODE REVIEW

**Review by y-ack — APPROVED** [review]

File: src/Views/page.js [hunk 1]
```diff
@@ -100,6 +101,21 @@ class PageView extends BaseView {
 			{icon:"✏️", label:"edit", href:"#editpage/"+page.id},
 			{icon:"🗂️", label:"childs", href:"#category/"+page.id},
 		])
+
+		if (!OPTS.has('dev')) {
+			const domain = Req.server.split("/")[0]
+			let href = undefined
+			if (Object.hasOwn(page.values, "share") && page.values.share) {
```
Comment (12Me21) [inline src/Views/page.js]: why are we checking page.values.share twice? (i would just use if (page.values.share))

Comment (ansxor) [reply src/Views/page.js]: it's to check if it's also a truthy value. if page.values.share won't throw an exception then i can also do that

File: src/navigate.js [hunk 2]
```diff
@@ -66,6 +66,7 @@ class ViewSlot {
 		for (let x of items) {
 			let a = document.createElement('a')
 			a.href = x.href
+			a.target = x['target'] ?? '_self'
```
Comment (12Me21) [inline src/navigate.js]: - should just be x.target || '_self' (i don't use ?? anywhere else in the code (..i hope i had a good reason for this))

File: src/Views/page.js (hunk 1 above)
Comment (ansxor) [reply src/Views/page.js]: ok yes i can do that, pushing a change

File: src/navigate.js (hunk 2 above)
Comment (ansxor) [reply src/navigate.js]: fineee

**Pull request merged**